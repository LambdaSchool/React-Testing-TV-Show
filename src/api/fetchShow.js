import axios from 'axios';

export const fetchShow = async () => {
    return axios
    .get("https://api.tvmaze.com/singlesearch/shows?q=stranger-things&embed=episodes")
    .then(res => {
        console.log("This is the fetchShow response", res);
        return res;
    })
    .catch( err => {
        console.log("Error generated by fetchShow API Component: ", err);
        return err;
    });
};