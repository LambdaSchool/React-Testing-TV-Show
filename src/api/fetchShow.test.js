import fetchShow from "./fetchShow";
import {render} from "@react-testing-library";

test("I know what I am doing", () => {
    expect(0).toEqual(1)
    // Fails because I am clueless
})

test("That app loads with correct data", () => {
    expect(fetchShow).toMatchSnapshot();
});

test("That I can even do something as basic and simple as loading the darned screen successfully", () => {
    render(<fetchShow />)
})

// welp this is absolutely it. I hate testing and its clear testing hates me. Please fail me for this project. I will spend the next 4 weeks studying react testing. I will make sure that i know how to do it. I just can not do it at this time and I am frustrating myself beyond belief even trying. People say react testing is easy. Not for me....