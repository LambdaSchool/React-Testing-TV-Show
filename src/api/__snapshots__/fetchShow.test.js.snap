// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`That app loads with correct data 1`] = `[Function]`;
